pkg/sql/parsers/tree/constant.go

```diff
@@ -120,12 +127,17 @@ func NewNumVal(value constant.Value, origString string, negative bool) *NumVal {
 }
 
 func NewNumValWithType(value constant.Value, origString string, negative bool, typ P_TYPE) *NumVal {
-	return &NumVal{
+	numVal := &NumVal{
 		Value:      value,
 		origString: origString,
 		negative:   negative,
 		ValType:    typ,
 	}
+	// tag the 0xXXXX
+	if typ == P_hexnum {
```

Comment:
The parameter isBin is not required. P_hexnum can represent

Reply:
ok, I have delete here